packages/components/src/core/Panel/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { StyledDrawer, StyledHeaderComponent } from "./style";
+import { DrawerProps } from "@mui/material";
+import PanelHeader from "./components/PanelHeader";
+import PanelHeaderClose, {
+  PanelHeaderCloseProps,
+} from "./components/PanelHeaderClose";
+
+export interface BasicPanelProps extends Omit<DrawerProps, "variant"> {
+  sdsType?: "basic"; // Discriminator
+  position?: "left" | "right";
+  width?: number | string;
+}
+
+export interface OverlayPanelProps extends Omit<DrawerProps, "variant"> {
+  sdsType?: "overlay"; // Discriminator
+  position?: "left" | "right" | "bottom";
+  width?: number | string;
+  headerComponent?: React.ReactNode;
+  closeButtonOnClick?: PanelHeaderCloseProps["onClick"];
+  closeButtonComponent?: PanelHeaderCloseProps["closeButtonComponent"];
+}
+
+// Discriminated Union
+export type PanelProps = BasicPanelProps | OverlayPanelProps;
+
+// Type guard to narrow the type
+function isOverlayPanelProps(props: PanelProps): props is OverlayPanelProps {
+  return props.sdsType === "overlay";
+}
+
+export const PANEL_BASIC_MIN_WIDTH = 240;
+export const PANEL_OVERLAY_MIN_WIDTH = 320;
+
+/**
+ * @see https://mui.com/material-ui/react-drawer/
+ */
+
+const Panel = React.forwardRef<HTMLDivElement, PanelProps>((props, ref) => {
+  const { children, sdsType = "basic", position = "left", width } = props;
+
+  const DrawerWidth = width
```

Comment:
feels like we could just use lowercase here, since it's just a regular variable! 
```suggestion
  const drawerWidth = width
```

Reply:
Oh, sure! 😯